### PR TITLE
[Dotenv] [DebugCommang] Ensure console commands respect `dotenv_path`

### LINKED ISSUE #58158: [Dotenv][DebugCommang]: Сommand ignores dotenv_path setting in composer.json
### Symfony version(s) affected

5.4

### Description

The console command `bin/console debug:dotenv` ignores the `dotenv_path` setting specified in the composer.json file.

### How to reproduce

Let's assume that `.env*` files should be located in the directory `<app_root>/config`

1. Set the .env file path in composer.json using dotenv_path:

```json
    "extra": {
        "runtime": {
            "dotenv_path": "config/.env"
        }
    }
```

2. Instantiate command:

```php
$projectDirectory = '/var/www/api';
new DebugCommand($_ENV['APP_ENV'], $projectDirectory);
```

3. Run the console command `bin/console dump:dotenv`
4. Observe that the command does not use the specified path.

### Possible Solution

Use the same logic as in the `dotenv:dump` command to determine the `dotenv_path`.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by scruwi**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | Fix #58158 
| License       | MIT

fix: ensure console commands respect `dotenv_path` setting from composer.json

- Updated logic to use the same approach as the `dotenv:dump` command for determining `dotenv_path`.

### CODE REVIEW

File: src/Symfony/Component/Dotenv/Command/DebugCommand.php [hunk 1]
```diff
@@ -50,7 +50,9 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             return 1;
         }
 
-        $filePath = $this->projectDirectory.\DIRECTORY_SEPARATOR.'.env';
+        $composerFile = $this->projectDirectory . '/composer.json';
+        $config = (is_file($composerFile) ? json_decode(file_get_contents($composerFile), true) : [])['extra']['runtime'] ?? [];
+        $filePath = $this->projectDirectory.\DIRECTORY_SEPARATOR.($config['dotenv_path'] ?? '.env');
```
Comment (xabbuh) [inline src/Symfony/Component/Dotenv/Command/DebugCommand.php]:
```suggestion
        $filePath = $this->projectDirectory.'/'.($config['dotenv_path'] ?? '.env');
```

using the constant is not necessary, let's replace it when we are at it

**Comment by xabbuh** [comment]
Can we add a test for it?

File: src/Symfony/Component/Dotenv/Tests/Command/DebugCommandTest.php [hunk 2]
```diff
@@ -167,10 +167,27 @@ public function testWarningOnPhpEnvFile()
...
 
-    private function executeCommand(string $projectDirectory, string $env): string
+    public function testLoadEnvFilesFromSubdirectory()
+    {
+        $projectDirectory = __DIR__.'/Fixtures/Scenario4';
+        file_put_contents($projectDirectory.'/composer.json', '{"extra":{"runtime":{"dotenv_path": "config/.env"}}}');
+
+        $output = $this->executeCommand($projectDirectory, 'dev', 'config/.env');
+
+        @unlink($projectDirectory.'/composer.json');
```
Comment (xabbuh) [inline src/Symfony/Component/Dotenv/Tests/Command/DebugCommandTest.php:177]: Why is this necessary? Why don't we store the `composer.json` file in the fixtures directory in the repo?

Comment (scruwi) [reply src/Symfony/Component/Dotenv/Tests/Command/DebugCommandTest.php:177]: I did it before, but some tests failed for some reason. Please look at the previous commmit.

**Comment by nicolas-grekas** [comment]
This concern was solved in #52638
It relies on a new feature so updating to 7.1 is required.
I don't think it's worth fixing in lower branches given this requires hacky solutions (bringing knowledge of composer.json here)
